fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java

```diff
@@ -1471,18 +1472,43 @@ private String mapToActualAppTable(final String appTable) {
         return appTable;
     }
 
-    private List<ResultsetRowData> fillDatatableResultSetDataRows(final String sql) {
+    private List<ResultsetRowData> fillDatatableResultSetDataRows(final String sql, final List<ResultsetColumnHeaderData> columnHeaders) {
         final List<ResultsetRowData> resultsetDataRows = new ArrayList<>();
+        final GenericResultsetData genericResultsetData = new GenericResultsetData(columnHeaders, null);
 
         final SqlRowSet rowSet = jdbcTemplate.queryForRowSet(sql); // NOSONAR
         final SqlRowSetMetaData rsmd = rowSet.getMetaData();
 
         while (rowSet.next()) {
-            final List<String> columnValues = new ArrayList<>();
+            final List<Object> columnValues = new ArrayList<>();
+
             for (int i = 0; i < rsmd.getColumnCount(); i++) {
                 final String columnName = rsmd.getColumnName(i + 1);
-                final String columnValue = rowSet.getString(columnName);
-                columnValues.add(columnValue);
+                if (columnHeaders.isEmpty()) {
+                    columnValues.add(rowSet.getString(columnName));
+                } else {
+                    final String colType = genericResultsetData.getColTypeOfColumnNamed(columnName);
+                    if ("DECIMAL".equalsIgnoreCase(colType)) {
+                        columnValues.add(rowSet.getBigDecimal(columnName));
+                    } else if ("DATE".equalsIgnoreCase(colType)) {
+                        columnValues.add(rowSet.getDate(columnName));
+                    } else if ("timestamp without time zone".equalsIgnoreCase(colType) // PostgreSQL
```

Comment:
@josehernandezfintecheandomx What do you think?

Reply:
@adamsaghy I can't include all the Date cases in one if, because in MySQL 8 we have an exception, Please see the image attached

<img width="1739" alt="Screen Shot 2022-07-24 at 19 42 10" src="https://user-images.githubusercontent.com/44206706/181264834-d9dac89b-6759-4fc9-9801-2871cf715bff.png">

Reply:
Please check all of my recommendations and i believe the issue will not be there anymore! ;)